restapi/endpoints_test.go

```diff
@@ -0,0 +1,197 @@
+package restapi
+
+import (
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/gavv/httpexpect/v2"
+	"github.com/spf13/viper"
+	"github.com/stretchr/testify/suite"
+	"github.com/uwblueprint/shoe-project/internal/database/migrations"
+	"github.com/uwblueprint/shoe-project/internal/database/models"
+	"github.com/uwblueprint/shoe-project/testutils"
+	"gorm.io/gorm"
+)
+
+type endpointTestSuite struct {
+	suite.Suite
+	endpoint *httpexpect.Expect
+	db       *gorm.DB
+	token    string
+}
+
+func (suite *endpointTestSuite) SetupSuite() {
+	db, err := testutils.MockDatabase()
+	if err != nil {
+		suite.Fail("error while creating database", err)
+	}
+	suite.db = db
+
+	// setup required for jwt authentication
+	viper.SetDefault("auth.jwt_key", "random")
+	viper.SetDefault("auth.jwt_expiry", "5m")
+	viper.SetDefault("auth.jwt_issuer", "endpoint_tests")
+
+	router, err := Router(db)
+	if err != nil {
+		suite.FailNow("error while creating router", err)
+	}
+
+	server := httptest.NewServer(router)
+	suite.endpoint = httpexpect.New(suite.T(), server.URL)
+}
+
+func (suite *endpointTestSuite) SetupTest() {
+	if err := migrations.CreateTables(suite.db); err != nil {
+		suite.Fail("error while creating tables", err)
+	}
+	if err := migrations.CreateSuperUser(suite.db); err != nil {
+		suite.Fail("error creating super user", err)
+	}
+
+	suite.token = suite.endpoint.POST("/login").
+		WithJSON([]models.User{{
+			Username: "admin",
+			Password: "root",
+		}}).
+		Expect().
+		Status(http.StatusOK).JSON().Object().Value("payload").String().Raw()
+}
+
+func (suite *endpointTestSuite) TearDownTest() {
+	if err := testutils.DropTables(suite.db); err != nil {
+		suite.Fail("error while dropping tables", err)
+	}
+}
+
+func (suite *endpointTestSuite) TestHealthCheck() {
+	suite.endpoint.GET("/health").
+		Expect().
+		Status(http.StatusOK).JSON().Object().Value("message").String().Equal("Hello World")
+}
+
+func (suite *endpointTestSuite) TestGetAllStories() {
+	json_story1 := []models.Story{
+		{
+			Title:    "The Little Prince",
+			Content:  "Children",
+			AuthorID: 1,
+		},
+	}
+
+	json_story2 := []models.Story{
+		{
+			Title:    "Hitchhiker's Guide to the Galaxy",
+			Content:  "Fiction",
+			AuthorID: 2,
+		},
+	}
+
+	//Directly add to mock DB
+	suite.db.Create(&json_story1)
+	suite.db.Create(&json_story2)
+
+	var response = suite.endpoint.GET("/stories").
+		Expect().
+		Status(http.StatusOK).
+		Status(http.StatusOK).JSON()
+
+	mock := `{
+		"payload": [
+		{
+			"author_id": 1,
+			"content": "Children",
+			"title": "The Little Prince"
+		},
+		{
+			"author_id": 2,
+			"content": "Fiction",
+			"title": "Hitchhiker's Guide to the Galaxy"
+		}
+		],
+		"status": "OK"
+	}`
+
+	//Verify mock matches response
+	response.Schema(mock)
```

Comment:
i was curious to see whether `.Schema` only validated the json schema (i.e. the shape of the json), so I tried to run the test with the following json:

```go
	mock := `{
		"payload": [
		{
			"author_id": 4,
			"content": "blablablabla",
			"title": "yes"
		},
		{
			"author_id": 5,
			"content": "",
			"title": "i dont exist"
		}
		],
		"status": "OK"
	}`
```

...and it still worked :( 

so `response.Schema(...)` only checks the shape of the response, but not the actual values. i still think it's valuable to check the actual values of the stories in the response - this would involve doing something like:

```
response.Value("payload").Array().Element(0).Object().Value("ID").Equal(1)
```

@hanlinc27  @dhillondeep what do u think?

Reply:
added that in :) should be all good now 👍